### ISSUE TITLE
[BUG][1.14.0] Sfge throwing error for managed package code

### ISSUE THREAD

**Issue by sposinadevops** [forcedotcom/code-analyzer#1988]
### Have you tried to resolve this issue yourself first?

- [x] I confirm I have gone through the above steps and still have an issue to report.

### Bug Description

When we try to run sfge engine on Managed package code, it is erroring with Unexpected token '(' . We understand managed package code is displayed as (Hidden) and that is where engine is throwing as unexpected token '('

### Output / Logs

```shell
The engine with name 'sfge' threw an unexpected error: Error: Salesforce Graph Engine encountered an error and couldn't complete analysis: | Graph engine encountered compilation errors. Fix the errors in C:\Users\extnhaley\Documents\Security Reports\sfdc-
prod\unpackaged\main\default\classes\APXTConga4_AccountFieldMetadataFactory.cls and retry., Caused by: | Invalid syntax at 1:1. (Unexpected token '('.) at RuntimeSfgeWrapper.invokeRunCommand (C:\Users\extnhaley\AppData\Local\sf\node_modules\@salesforce\plugin-code-
analyzer\node_modules\@salesforce\code-analyzer-sfge-engine\dist\sfge-wrapperjs:78:19) at process.processTicksAndRejections (node:internal/process/task_queues:105:5) at async SfgeEngine.runRules (C:\Users|extnhaley\AppData\Local\sf\node_modules)@salesforce\plugin-code-
analyzer\node_modules\@salesforce\code-analyzer-sfge-engine\dist\engine.js:68:29) at async
CodeAnalyzer.runEngineAndValidateResults (C:\Users\extnhaley\AppData\Local\sf/node_modules\@salesforce\plugin-
code-analyzer\node_modules\@salesforce\code-analyzer-core\dist\code-analyzer.js:389:35) at async
C:\Users\extnhaley\AppData\Local\sf\node_modules)@salesforce\plugin-code-
analyzer \node_modules\@salesforce\code-analyzer-core\dist\code-analyzer.js:276:29 at async Promise.all (index 1) at async CodeAnalyzer.run (C:\Users\extnhaley\AppData\Local\sf\node_modules\@salesforce\plugin-code-
analyzer\node_modules\@salesforce\code-analyzer-core\dist\code-analyzer.js:283:38) at async RunAction.execute (file:///C:/Users/extnhaley/AppData/Local/sf/node_modules/@salesforce/plugin-code-
analyzer/lib/lib/actions/RunAction.js:39:25) at async RunCommand.run
(file:///C:/Users/extnhaley/AppData/Local/sf/node_modules@salesforce/plugin-code-analyzer/lib/commands/code-
analyzer/run.js:86:9) at async RunCommand._run
(C:\Users\extnhaley\AppData\Local\sf\node_modules\@salesforce\plugin-code-
analyzer\node_modules\@oclif\core\lib\command.js:182:22)
```

### Steps To Reproduce

Run sfge engine on Managed package code

### Expected Behavior

Just want to understand in future, if change will be made to disregard this behavior and continue with processing (or) shouldn't we run engine on managed package code? 

### Operating System

mac os seqioa 15.7.3

### Salesforce CLI Version

salesforce/cli/2.117.6 darwin-arm64 node-v22.21.1

### Code Analyzer Plugin (code-analyzer) Version

5.9.0node

### Additional Context (Screenshots, Files, etc)

<img width="853" height="359" alt="Image" src="https://github.com/user-attachments/assets/bc307889-5ee6-4a6f-8158-93d18d43dd34" />

### Urgency

Moderate

**Comment by aruntyagiTutu** [comment]
Hi @sposinadevops , thanks for reporting this. The issue occurs because SFGE tries to compile managed package classes that show as (Hidden) in your org. The engine sees the literal text (Hidden) and fails to parse it as valid Apex.
Workaround: Exclude managed package code from your scan. Use the --project-dir flag to target only your custom code directories, or use --category to avoid graph-based rules (though you'd lose SFGE's insights).
Example:
sf code-analyzer run --target "force-app/main/default/classes" --engine sfge

please let me know if you are still facing issues when using it on target folder

**Comment by sposinadevops** [comment]
Hi @aruntyagiTutu Please find the folder structure below. Customer have all classes including managed and their own classes under single folder called classes

<img width="436" height="294" alt="Image" src="https://github.com/user-attachments/assets/fd1d5754-904c-4752-913f-0c7b574c5ed5" />
